README.md

```diff
@@ -1,9 +1,82 @@
-# sample
+# pedantic
 
-A library for Dart developers. It is awesome.
+This package serves three purposes:
+
+ - It documents how Dart static analysis is used internally at Google,
+   including best practices for the code we write. The documentation is
+   this `README.md`.
+ - It contains a corresponding sample `analysis_options.yaml`.
+ - It contains occasional small snippets of Dart code that are used in
+   implementing those best practices.
+
+Note that everything here fits within the guidelines set out in
+[Effective Dart](https://www.dartlang.org/guides/language/effective-dart).
+You could think of that document as the _design_ and this package as one
+possible partial _implementation_.
+
+## Using Static Analysis
+
+Here is how static analysis is used internally at Google:
+
+ - By default we disallow checking in code with any errors, warnings, or hints.
+   - The `TODO` hint is a permanent exception.
+   - Deprecation hints are a permanent exception. Deprecations are handled
+     separately on a case by case basis.
+   - `unnecessary_no_such_method`, `unused_element`, `unused_field` and
+     `unused_local_variable` are allowed.
+   - When a new SDK version adds new errors, warnings or hints, we either clean
+     up everything before switching SDK version or maintain a whitelist of
+     allowed violations so it can be gradually cleaned up.
+ - Lints are considered and enabled on a case by case basis. When enabling a
+   lint we first clean up all pre-existing violations. After it's enabled, any
+   attempt to check in a further violation will be blocked.
+
+The currently enabled lints are:
```

Comment:
Rather than duplicate all of this (so that changes have to be made in two places), consider just pointing readers to the sample analysis options file.

Reply:
Thanks! I agree with de-duping, but I think the md file is a nicer place to talk about reasons for not enforcing particular lints. So I put the enforced lints just in the yaml file and the not enforced lints just in the README. What do you think, please?

Thanks :)

Reply:
I think that's fine. You can always change it if you discover that it isn't working well for you :-) After all, you're the one who'll be maintaining this (at least for a while).

Reply:
Quite :)

Thanks!